Added Student List with Section Assignment, Persistence, and Tests

Maharshi1208-conestoga opened the pull request:
### Summary
- Implemented StudentList UI with pagination
- Added individual and bulk section assignment (A/B/C)
- Section data persisted using `localStorage`
- Logs view + clear functionality
- Added `studentData.ts` as data layer
- Fully tested with `StudentList.test.tsx`

Maharshi1208 committed:
Add StudentProfile component and test

Maharshi1208 committed:
✅ Implemented StudentList with test cases and persistent section assignment

Maharshi1208-conestoga commented:
> hi Maharshi,
> 
> the question here is still the same as before. What objective are you working on.
> 
> this student profile looks a whole lot like features that is already being worked on by others.
> 
> The quality of software is strongly tied to organization. Part of this means that we are all developing things that are well integrated. We need to be sure that you are expanding on functionality but not competing with others.
> 
> so please tell me the objective and the issue along with integration.



Hi, Prof.,

Objective:
My primary goal is to automate user enrollment across multiple locations in the SENG8130 system. This effort aligns with the OKR:
"Enable bulk and manual group/section assignments."
This objective is documented in our shared planning board, where I am listed as the feature owner.
 Purpose of the Work (Key Result):
This PR introduces the student assignment interface, supporting:
Manual assignment using a dropdown menu beside each student

Bulk assignment using checkboxes with a shared dropdown for section selection

Integration:
Extends the /students route within the existing Enrollment module

Utilizes and updates the shared studentData.ts mock data

Avoids duplication or overlap with other active features — focuses strictly on assignment interactions (not listing, filtering, or evaluation logic)

Fully compatible with the existing layout and navigation patterns

Next Steps:
Persistence and logging (as outlined in the success metrics) will be implemented in a follow-up PR.
Please let me know if anything needs adjustment or better alignment with parallel efforts.

Thanks.

achow-conestogac commented on src/App.tsx:
```diff
@@ -4,12 +4,15 @@ import logo from './assets/logo.png';
       <Routes>
         <Route path="/enrollment" element={<EnrollmentDashboard />} />
         <Route path="/evaluation" element={<EvaluationCard />} />
+        <Route path="/studentlist" element={<StudentList />} /> {/* ✅ Add this route */}
```
The student list is something that only concerns `enrollment`.  it should not be a separate route.

achow-conestogac commented on src/Enrollment/Student/Pages/StudentList/StudentList.tsx:
```diff
@@ -10,13 +9,17 @@ const StudentList: React.FC = () => {
+  const [selectedStudents, setSelectedStudents] = useState<string[]>([]);
+  const [bulkSection, setBulkSection] = useState<string>("");
+  const [logsVisible, setLogsVisible] = useState(false);
+const logs = JSON.parse(localStorage.getItem("logs") || "[]");
```
there is a `LocalStorageService` implemented. use that for access to the `localStorage`

achow-conestogac commented on src/Enrollment/Student/Pages/StudentList/StudentList.tsx:
```diff
@@ -54,6 +134,37 @@ const StudentList: React.FC = () => {
             <th className="px-4 py-3 text-right">Actions</th>
           </tr>
         </thead>
+        <tbody>
+          {students.map(student => (
+            <tr key={student.id} className="border-t">
+              <td className="px-4 py-2">
+                <input
+                  type="checkbox"
+                  checked={selectedStudents.includes(student.id)}
+                  onChange={() => handleCheckboxChange(student.id)}
+                />
+              </td>
+              <td className="px-4 py-2">{student.name}</td>
+              <td className="px-4 py-2">{student.email}</td>
+              <td className="px-4 py-2">{student.role}</td>
+              <td className="px-4 py-2">{student.section}</td>
+              <td className="px-4 py-2">{student.group}</td>
+              <td className="px-4 py-2 text-right">
+                <select
+                  onChange={(e) => handleManualAssign(student.id, e.target.value)}
+                  defaultValue=""
+                  className="border rounded px-2 py-1"
+                  data-testid={`manual-select-${student.id}`}
+                >
+                  <option value="" disabled>Assign Section</option>
+                  <option value="A">Section A</option>
+                  <option value="B">Section B</option>
+                  <option value="C">Section C</option>
+                </select>
+              </td>
+            </tr>
+          ))}
+        </tbody>
```
This looks like something that is being implemented by @Shiva30jun 

have you guys discussed the integration between your two teams?

achow-conestogac commented on src/Enrollment/Student/Pages/StudentProfile/StudentProfile.test.tsx:
```diff
@@ -0,0 +1,66 @@
+import StudentProfile from "./StudentProfile";
+
+// Mock student service
+jest.mock("../../Services/findStudentById", () => ({
```
do not use jest mock.

please make sure that you are using proper dependency inversions in order to enable testing.

achow-conestogac commented on src/Enrollment/Student/Pages/StudentProfile/StudentProfile.test.tsx:
```diff
@@ -0,0 +1,66 @@
+  })),
+}));
+
+jest.mock("../../Services/deleteStudentById", () => ({
```
to go further, what are you mocking here. that directory does not exist.

Maharshi1208 committed:
✅ Refactored routing under /enrollment, integrated student profile view, replaced localStorage

Maharshi1208 committed:
Removed package-lock.json to resolve conflict — not part of this PR

Maharshi1208 committed:
Resolved merge conflict in package-lock.json by regenerating it cleanly

Maharshi1208-conestoga committed:
Merge branch 'main' into student-profile

Maharshi1208 committed:
✅ Added 10 test cases for StudentProfile component

Maharshi1208-conestoga commented:
Hi Professor,

I’ve completed the required updates and pushed the latest changes to the student-profile branch.
Here’s a summary of what’s been addressed:

Resolved previous code integration and routing issues

Refactored StudentProfile.tsx using dependency injection

Created a container component for better testability

Added and tested 10+ unit test cases in StudentProfile.test.tsx

Handled routing and fallback conditions (invalid/missing student IDs)

Successfully pushed and rebased with the latest upstream code

Kindly review and let me know if anything else is needed.
Thank you!
![screenshot](https://github.com/user-attachments/assets/67194213-a9ba-4323-9a0d-4dd4bb04e3d4)
